### PR TITLE
Store allele synonyms

### PR DESCRIPTION
**Pull request by helensch**

### CODE REVIEW

File: sql/table.sql [hunk 1]
```diff
@@ -256,6 +256,29 @@ create table variation_synonym (
...
+*/
+
+CREATE TABLE allele_synonym (
+  allele_synonym_id int(10) unsigned NOT NULL AUTO_INCREMENT,
+  variation_id      int(10) unsigned NOT NULL,
+  hgvs              varchar(600) NOT NULL,
+  name              varchar(255) NOT NULL,
+  PRIMARY KEY (allele_synonym_id),
+  UNIQUE KEY variation_name_idx (variation_id, name),
+  KEY hgvs_idx (hgvs)
```
Comment (sarahhunt) [inline sql/table.sql]: We also need an index on the name to enable look up by CA id.

File: sql/table.sql [hunk 2]
```diff
@@ -256,6 +256,29 @@ create table variation_synonym (
...
+@column hgvs                  HGVS of variant e.g. NC_000001.11:g.230710048A>G
+@column name                  Name of the allele synonym e.g. CA127784
+
+@see variation
+*/
+
+CREATE TABLE allele_synonym (
+  allele_synonym_id int(10) unsigned NOT NULL AUTO_INCREMENT,
+  variation_id      int(10) unsigned NOT NULL,
+  hgvs              varchar(600) NOT NULL,
```
Comment (sarahhunt) [inline sql/table.sql]: We refer to this value as hgvs_genomic in the transcript_variation table; it may be useful to be consistent.

**Comment by sarahhunt** [comment]
It is customary to use patch 'a' for the schema version change. I don't think anything is dependent on this.

File: sql/table.sql (hunk 2 above)
Comment (helensch) [reply sql/table.sql]: Is the only genomic hgvs going to be used?

File: sql/table.sql (hunk 1 above)
Comment (helensch) [reply sql/table.sql]: I'll add  that.  Should I remove the hgvs index?

**Comment by helensch** [comment]
Do you want me to rename this to patch b?
I'll add the patch to the table.sql

File: sql/table.sql (hunk 2 above)
Comment (sarahhunt) [reply sql/table.sql]: I can't currently think of a reason to use any other.

File: sql/table.sql (hunk 1 above)
Comment (sarahhunt) [reply sql/table.sql]: Do you need it for populating the variant summary info on the website?

**Comment by sarahhunt** [comment]
It may be clearer to call this patch b.

File: sql/table.sql (hunk 2 above)
Comment (helensch) [reply sql/table.sql]: I'll change the name. The ClinGen Allele Registry supports search using HGVS expressions for a transcripts, genomic and protein sequences .

**Comment by helensch** [comment]
Replaced by #231 that includes schema patch and addition of allele_synonym table.

**Pull request closed without merging**